integrations/kube-agent-updater/cmd/teleport-kube-agent-updater/main.go

```diff
@@ -108,16 +113,36 @@ func main() {
 		os.Exit(1)
 	}
 
-	// TODO: replace those mocks by the real thing
-	versionGetter := version.NewGetterMock("12.0.3", nil)
-	imageValidators := []img.Validator{
-		img.NewImageValidatorMock("mock", true, img.NewImageRef("", "", "", "")),
+	versionServerUrl, err := url.Parse(versionServer + "/" + versionChannel)
+	if err != nil {
+		ctrl.Log.Error(err, "failed to pasre version server URL, exiting")
+		os.Exit(1)
+	}
+	versionGetter := version.NewBasicHTTPVersionGetter(versionServerUrl)
+	maintenanceTriggers := maintenance.Triggers{
+		maintenance.NewBasicHTTPMaintenanceTrigger("critical update", versionServerUrl),
+		maintenance.NewUnhealthyWorkloadTrigger("unhealthy pods", mgr.GetClient()),
+		maintenance.NewWindowTrigger("maintenance window", mgr.GetClient()),
+	}
+
+	var imageValidators img.Validators
+	if insecureNoVerify {
+		ctrl.Log.Info("INSECURE: Image validation disabled")
+		imageValidators = append(imageValidators, img.NewInsecureValidator("insecure always verify"))
+	} else {
+		validator, err := img.NewCosignSingleKeyValidator(teleportProdOCIPubKey, "cosign signature validator")
```

Comment:
Should we support two keys? To allow key rotation
This was probably discussed previously, sorry if so

Reply:
We can chain validators: you only need one happy validator to accept the image. This way, I we want to introduce new keys, new validation methods, or do a key rotation, we can add a second validator with the new key and support a transition period.